test/handler_test.cpp

```diff
@@ -70,3 +70,18 @@ TEST(StaticHandlerTest, StaticClassTest){
 
 }
 
+TEST(ProxyHandlerTest, ProxyClassTest) {
+  std::unique_ptr<ProxyHandler> handler(new ProxyHandler); 
+
+  handler->SetHost("ucla.edu"); 
```

Comment:
By using the setHost public method, the public API changed which goes aganist the spec I believe. However even if that were not the case, by using the public method to initialize/set the host, we are sidestepping testin gthe ProxyHandler Init function. Might as well just use the Init function to set the host right? Knocks two birds with one stone